.github/templates/main.yml

```diff
@@ -115,14 +115,24 @@ steps:
 #@ end
 
 #@ def uploadPackageArtifacts():
-#@ for pkgName in [ 'Realm.Fody', 'Realm' ]:
-#@ finalPkgName = pkgName + ".${{ steps.find-nupkg-version.outputs.package_version }}"
-  - name: #@ "Store artifacts for " + pkgName
-    uses: #@ actionUploadArtifact
+#@ pkgNames = [ 'Realm.Fody', 'Realm' ]
+#@ relativePkgPaths = []
+#@ for pkgName in pkgNames:
+#@  finalPkgName = pkgName + ".${{ steps.find-nupkg-version.outputs.package_version }}" 
+#@  relativePkgPaths.append("Realm/packages/" + finalPkgName + ".*nupkg")
+#@ end
+#@ pkgNames.append("ExtractedChangelog.md")
+#@ relativePkgPaths.append("Realm/Realm/ExtractedChangelog.md")
+#@ for i in range(0, len(pkgNames)):
+#! get the filename with no extension
+#@ finalPkgName = relativePkgPaths[i][(relativePkgPaths[i].rfind("/") + 1) : len(relativePkgPaths[i])]
+#@ finalPkgName = finalPkgName[0 : finalPkgName.rfind(".")]
```

Comment:
This seems like quite a lot of complexity to just add an extra step. Since `uploadPackageArtifacts` is only used once, it's better to just add a step after it to upload the release notes.

Reply:
Do you prefer it in the fuction or right after it's used? I'd expect in the function.

Reply:
Since the function is called `uploadPackageArtifacts`, I'd keep it focused on uploading only the package artifacts. We can have a final step, after `template.replace(buildUnityPackage())` that uploads the release notes. We can also extract a function for uploading artifacts - something like:

```yaml
#@ def uploadArtifacts(name, path):
  - name: #@ "Store artifacts: " + name
    uses: #@ actionUploadArtifact
    with:
      name: #@ name
      path: #@ "${{ github.workspace }}/" + path
      retention-days: ${{ github.event_name != 'pull_request' && 30 || 1 }}
#@ end
```

We can then use it in `buildUnityPackage`, `uploadPackageArtifacts` and all the other places where we upload artifacts. That can be done in a follow-up PR though, let's not bloat this one with unrelated improvements.